File: Python/program/Storage.py
from Komponen import Komponen
from Connectable import Connectable

class Storage(Komponen, Connectable):
    def __init__(self, merk=None, nama=None, kapasitasGB=None, kecepatanMBps=None, interfaceType=None):
        Komponen.__init__(self, merk, nama)
        self.kapasitasGB = kapasitasGB
        self.kecepatanMBps = kecepatanMBps
        self.interfaceType = interfaceType

    # Getter dan Setter
    def setKapasitasGB(self, kapasitasGB):
        self.kapasitasGB = kapasitasGB

    def setKecepatanTransfer(self, kecepatanMBps):
        self.kecepatanMBps = kecepatanMBps

    def setInterfaceType(self, interfaceType):
        self.interfaceType = interfaceType

    def getKapasitasGB(self):
        return self.kapasitasGB

    def getKecepatanTransfer(self):
        return self.kecepatanMBps

    def getInterfaceType(self):
        return self.interfaceType

    # Override metode dari Connectable
    def getConnectionType(self):
        return self.interfaceType

    def displayInfo(self):
        super().displayInfo()
        print(f"Storage: {self.nama} ({self.merk}) - {self.kapasitasGB} GB, {self.kecepatanMBps} MBps, Interface: {self.interfaceType}")

File: Java/Program/Storage.java
public abstract class Storage extends Komponen implements Connectable {
    private int kapasitasGB;
    private int kecepatanMBps;
    private String interfaceType;
    
    public Storage() {}
    
    public Storage(String merk, String nama, int kapasitasGB, int kecepatanMBps, String interfaceType) {
        super(merk, nama);
        this.kapasitasGB = kapasitasGB;
        this.kecepatanMBps = kecepatanMBps;
        this.interfaceType = interfaceType;
    }
    
    // Setter 
    public void setKapasitasGB(int kapasitasGB) {this.kapasitasGB = kapasitasGB;}
    public void setKecepatanMBps(int kecepatanMBps) {this.kecepatanMBps = kecepatanMBps;}
    public void setInterfaceType(String interfaceType) {this.interfaceType = interfaceType;}
    
    // Getter
    public int getKapasitasGB() {return kapasitasGB;}
    public int getKecepatanMBps() {return kecepatanMBps;}
    public String getInterfaceType() {return interfaceType;}
    
    //  Connectable interface
    @Override
    public String getConnectionType() {
        return interfaceType;
    }
    
    @Override
    public void displayInfo() {
        System.out.println("Storage: " + getNama() + " (" + getMerk() + ") - " + kapasitasGB + " GB, " + 
                            kecepatanMBps + " MBps, Interface: " + interfaceType);
    }
}